info-compiler/src/main/java/com/google/sps/webcrawler/WebCrawler.java

```diff
@@ -193,9 +227,9 @@ public String handleResponse(final HttpResponse response) throws IOException {
   /**
    * Waits for the required crawl delay to pass if necessary and makes a note of the required
    * crawl delay. Returns true if the aforementioned process succeeded. {@code grant} is expected
-   * to non-null.
+   * to non-null. This method is made public for testing purposes.
    */
-  private boolean waitForAndSetCrawlDelay(Grant grant, String url) {
+  public boolean waitForAndSetCrawlDelay(Grant grant, String url) {
```

Comment:
I think  you should be able to just make these methods "protected" instead of public if you wanted, since the tests fall in the same package, technically

Reply:
Oh sorry, I think I actually meant default (unless you actually need the additional functionality of "protected"):
https://www.tutorialspoint.com/what-are-the-differences-between-protected-and-default-access-specifiers-in-java#:~:text=What%20are%20the%20differences%20between%20protected%20and%20default%20access%20specifiers%20in%20Java%3F,-JavaObject%20Oriented&text=The%20Protected%20access%20specifier%20is,visible%20in%20the%20same%20package.

This also applies to variables as well

Reply:
Ah ok! Thank you